packages/editor/src/components/post-sync-status/index.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { ToggleControl, PanelRow } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+export default function PostSyncStatus() {
+	const { editPost } = useDispatch( editorStore );
+	const { meta, postType } = useSelect( ( select ) => {
+		const { getEditedPostAttribute } = select( editorStore );
+		return {
+			meta: getEditedPostAttribute( 'meta' ),
+			postType: getEditedPostAttribute( 'type' ),
+		};
+	}, [] );
+	if ( postType !== 'wp_block' ) {
+		return null;
+	}
+	const onUpdateSync = ( syncStatus ) =>
+		editPost( {
+			meta: {
+				...meta,
+				wp_block:
+					syncStatus === 'unsynced'
+						? { sync_status: syncStatus }
+						: null,
+			},
+		} );
+	const syncStatus = meta?.wp_block?.sync_status;
+	const isFullySynced = ! syncStatus;
+
+	return (
+		<PanelRow className="edit-post-sync-status">
+			<span>{ __( 'Syncing' ) }</span>
+			<ToggleControl
+				__nextHasNoMarginBottom
+				label={
+					isFullySynced ? __( 'Fully synced' ) : __( 'Not synced' )
+				}
+				checked={ isFullySynced }
+				onChange={ () => {
+					onUpdateSync(
+						syncStatus === 'unsynced' ? 'fully' : 'unsynced'
+					);
+				} }
+			/>
```

Comment:
It might be worth considering the consequences of this option and conveying them to the user in some help text, or using a warning modal.

When you change from unsynced to synced, the change will only apply to newly inserted patterns.

When you change from synced to unsynced, existing synced patterns will become unsynced when you edit them.

The latter part might be confusing for users as they'll have unexpected unsaved changes. (it might especially be confusing on sites with multiple users where one user changed the option, and a second user encounters the unsaved changes).

Reply:
I guess an option is to also remove the code that auto-converts the sync status. That might be the preferred option.

Reply:
> I guess an option is to also remove the code that auto-converts the sync status. 

We currently don't have any way to detect that a block used to be synched but was switched to unsynched so I am not sure how we would do this?

Reply:
@glendaviesnz I saw this code is added to the reusable block edit function in this PR:
```js
	if (
		hasResolved &&
		record?.meta?.sync_status === 'unsynced' &&
		blocks.length > 0
	) {
		replaceBlocks(
			clientId,
			blocks.map( ( block ) => cloneBlock( block ) )
		);
	}
```

My understanding is that it auto-converts a previously 'synced' but now 'unsynced' pattern. 

The suggestion is to remove this code so that existing blocks are unaffected by a switch from synced to unsynced. By doing that switching between synced and unsynced or back again works the same way - it'll only affect newly inserted patterns.